mock/mock.go

```diff
@@ -923,56 +923,75 @@ func (args Arguments) Is(objects ...interface{}) bool {
 	return true
 }
 
+type outputRenderer func() string
+
 // Diff gets a string describing the differences between the arguments
 // and the specified objects.
 //
 // Returns the diff string and number of differences found.
 func (args Arguments) Diff(objects []interface{}) (string, int) {
 	// TODO: could return string as error and nil for No difference
 
-	output := "\n"
+	var outputBuilder strings.Builder
 	var differences int
 
 	maxArgCount := len(args)
 	if len(objects) > maxArgCount {
 		maxArgCount = len(objects)
 	}
 
-	for i := 0; i < maxArgCount; i++ {
+	outputRenderers := []outputRenderer{}
+
+	for j := 0; j < maxArgCount; j++ {
+		i := j
```

Comment:
What's the difference between i and j?

Reply:
We close over the loop variable in the renderers, so this is to avoid the pre-1.22 loopvar behavior issues. Can definitely drop depending on support requirements

Reply:
Ah of course we do. I mistook it for an artefact from the removal of the outer loop. Could you make it into `i := i`? It is legal and I think more clearly communicates to most gophers that you're deliberately re-scoping the variable.